### PR TITLE
support images in chat

### LINKED ISSUE #1314: Chat control should support images
### Description

The chat control of #1092 is really cool! It would be a good addition to support images in the messages of the chat. This would be especially useful since now LLMs start supporting image vision and image generation. I could create cool demos around this chat control.

Florian:
People are also using chat demos to visualize tables inside their chat. This could also be a nice addition.

### Acceptance Criteria

- [ ] Ensure new code is unit tested, and check code coverage is at least 90%.
- [ ] Create related issue in taipy-doc for documentation and Release Notes.
- [ ] Check if a new demo could be provided based on this, or if legacy demos could be benefit from it.
- [ ] Ensure any change is well documented.

### Code of Conduct

- [X] I have checked the [existing issues](https://github.com/Avaiga/taipy/issues?q=is%3Aissue+).
- [ ] I am willing to work on this issue (optional)

### PR DESCRIPTION
**Pull request by FredLL-Avaiga**
resolves #1314

## What type of PR is this? (check all applicable)

- [x] Refactor
- [x] Feature
- [x] Bug Fix
- [x] Optimization
- [ ] Documentation Update

## Description

Support images in chat 
- `allow_send_images` True by default
- through data URL
- limit size and notify user if size too big
- examples

## Related Tickets & Documents
- Related Issue #1314

## How to reproduce the issue

```py
from taipy.gui import Gui, Icon

msgs = [
    ["1", "msg 1", "Alice", None],
    ["2", "msg From Another unknown User", "Charles", None],
    ["3", "This from the sender User", "taipy", "./beatrix-avatar.png"],
    ["4", "And from another known one", "Alice", None],
]
users = [
    ["Alice", Icon("./alice-avatar.png", "Alice avatar")],
    ["Charles", Icon("./charles-avatar.png", "Charles avatar")],
    ["taipy", Icon("./beatrix-avatar.png", "Beatrix avatar")],
]


def on_action(state, id: str, payload: dict):
    (reason, varName, text, senderId, imageData) = payload.get("args", [])
    msgs.append([f"{len(msgs) +1 }", text, senderId, imageData])
    state.msgs = msgs


page = """
<|{msgs}|chat|users={users}|allow_send_images|>
"""

if __name__ == "__main__":
    Gui(page).run(title="Chat - Images")

```

## Checklist

- [x] Does this solution meet the acceptance criteria of the related issue?
- [ ] Is the related issue checklist completed?
- [x] Does this PR adds unit tests for the developed code? If not, why?
- [ ] End-to-End tests have been added or updated?
- [ ] Was the documentation updated, or a dedicated issue for documentation created? (If applicable)
- [ ] Is the release notes updated? (If applicable)

### CODE REVIEW

File: doc/gui/examples/controls/chat_images.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
+from taipy.gui import Gui, Icon
```
Comment (FabienLelaquais) [inline doc/gui/examples/controls/chat_images.py:21]: Header is missing.

File: doc/gui/examples/controls/chat_messages.py [hunk 2]
```diff
@@ -26,21 +26,20 @@
...
 
 
-Gui(
-    """
-<|toggle|theme|>
-# Test Chat
+page="""
 <|1 1 1|layout|
-<|{msgs}|chat|users={users}|show_sender={True}|>
+<|{msgs}|chat|users={users}|show_sender={True}|don't allow_send_images|>
```
Comment (FabienLelaquais) [inline doc/gui/examples/controls/chat_messages.py]: Why not *allow* images in this example? That would make sense...

File: taipy/gui/viselements.json [hunk 3]
```diff
@@ -1815,8 +1815,14 @@
                     {
                         "name": "max_file_size",
                         "type": "int",
-                        "default_value": "1024 * 1024",
-                        "doc": "The maximum allowable file size, in bytes, for files uploaded to a chat message.\nThe default is 1 MB."
+                        "default_value": "0.8 * 1024 * 1024",
+                        "doc": "The maximum allowable file size, in bytes, for files uploaded to a chat message.\nThe default is 0.8 MB."
```
Comment (FabienLelaquais) [inline taipy/gui/viselements.json:1819]: 0.8? I didn't see that coming!

**Review by FabienLelaquais — DISMISSED** [review]
👍

File: taipy/gui/viselements.json (hunk 3 above)
Comment (FredLL-Avaiga) [reply taipy/gui/viselements.json:1819]: 750Ko OK 
900 Ko not OK

**Review by namnguyen20999 — APPROVED** [review]
Lgtm

**Review by dinhlongviolin1 — APPROVED** [review]
LGTM

File: doc/gui/examples/controls/chat_messages.py [hunk 4]
```diff
@@ -12,10 +12,10 @@
 from taipy.gui import Gui, Icon
 
 msgs = [
-    ["1", "msg 1", "Alice"],
-    ["2", "msg From Another unknown User", "Charles"],
-    ["3", "This from the sender User", "taipy"],
-    ["4", "And from another known one", "Alice"],
+    ["1", "msg 1", "Alice", None],
```
Comment (FabienLelaquais) [inline doc/gui/examples/controls/chat_messages.py:15]: Do we really need these 'image' values?

**Review by FabienLelaquais — APPROVED** [review]
👍

File: doc/gui/examples/controls/chat_messages.py (hunk 4 above)
Comment (FredLL-Avaiga) [reply doc/gui/examples/controls/chat_messages.py:15]: I guess we need to create a column if we want to support image

**Pull request merged**